narwhal/config/src/lib.rs

```diff
@@ -407,6 +407,8 @@ pub struct WorkerInfo {
     pub transactions: Multiaddr,
     /// Address to receive messages from other workers (WAN) and our primary.
     pub worker_address: Multiaddr,
+    /// Optional alternative address at which workers can receive messages from their own primary.
```

Comment:
Probably want to adjust the comment to the field above to clarify the difference of the two fields.

Reply:
Reworded this a bit, let me know if you think something is still unclear.

Reply:
Is it optional because we can only use this address when the worker/primary are co-located? If so, maybe you can specify that the "address should be used by a primary to talk to its own workers if co-located" in this line. And then in the line above you can mention that `worker_address` is used to "receive messages from other workers (WAN), other primaries and our primary IFF it is not co-located with it's worker"

Reply:
It's optional because you don't have to specify one. It will use the regular `worker_address` if you don't (which is still documented as being for "other workers (WAN) and our primary", just not "preferentially used" as the internal address is).

I don't think the docs should specify how this "should" be used because you can use it however you want, operationally, depending on what your routing constraints are.

Your rewording for the first line is inaccurate because colocation or lack thereof is not implied by these configs, and I don't think the "IFF" is actually an accurate representation of how the address is selected.

Reply:
Clearly the confusion indicates this could be worded better but I'm struggling to come up with anything that is more clear. I could change "to talk" to "when talking," but that doesn't seem to capture some of the confusion.

Was there anything in the specific wording that led you to believe co-location was a relevant factor in how these addresses are used (in which case I can explicitly try to remove that implication from the wording), or did that understanding just come from external context around this PR?